Sources/SWBLLBuild/LowLevelBuildSystem.swift

```diff
@@ -25,7 +21,41 @@ public import SWBLibc
 #endif
 
 // Filesystem adaptors for SWBLLBuild.FileSystem.
-extension SWBUtil.FileInfo: SWBLLBuild.FileInfo {}
+extension SWBUtil.FileInfo: SWBLLBuild.FileInfo {
+
+    public init(_ statBuf: stat) {
+        // This should be remove from llbuild FileInfo protocol as it just not needed, would also be nice to remove the stat requirement too.
+        preconditionFailure()
+    }
+    
+    public var statBuf: stat {
+        var statBuf: stat = stat()
+        #if canImport(Darwin)
+        statBuf.st_dev = self.deviceID
+        statBuf.st_ino = self.iNode
+        statBuf.st_mode = self.permissions
+        statBuf.st_size = self.size
+        statBuf.st_mtimespec.tv_sec = Int(self.modificationTimestamp)
+        statBuf.st_mtimespec.tv_nsec = self.modificationNanoseconds
+        #elseif os(Windows)
+        statBuf.st_dev = UInt32(self.deviceID)
+        statBuf.st_ino = UInt16(self.iNode)
+        statBuf.st_mode = self.permissions
+        statBuf.st_size = Int32(self.size)
+        statBuf.st_mtime = self.modificationTimestamp
+        #elseif canImport(Glibc) || canImport(Musl) || canImport(Android)
+        statBuf.st_dev = UInt(self.deviceID)
+        statBuf.st_ino = UInt(self.iNode)
+        statBuf.st_mode = UInt32(self.permissions)
```

Comment:
I forgot `canImport(Glibc)` is true on FreeBSD. However, mode_t is UInt16 there.

Maybe use this instead?

```swift
statBuf.st_mode = numericCast(self.permissions)
```

With that change I was able to successfully build for FreeBSD with the latest main + this PR.